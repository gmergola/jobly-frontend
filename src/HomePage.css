.Home-container {
  text-align: center;
}

.Home-title {
  margin-top: 40px;
}

.description {
  margin-bottom: 40px;
}