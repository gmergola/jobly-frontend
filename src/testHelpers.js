import React from "react";
import TokenContext from "./TokenContext";

const demoToken = 'TOKEN';

const TokenProvider = ({ children, currentToken = demoToken }) => (
  <TokenContext.Provider value={{currentToken}}>
    {children}
  </TokenContext.Provider>
);

export { TokenProvider };
