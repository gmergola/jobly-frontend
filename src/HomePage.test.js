import React from "react";
import { render } from "@testing-library/react";
import HomePage from "./HomePage";
import { MemoryRouter } from "react-router";
import { TokenProvider } from "./testHelpers";


it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <TokenProvider>
        <HomePage />
      </TokenProvider>
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <TokenProvider>
        <HomePage />
      </TokenProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});