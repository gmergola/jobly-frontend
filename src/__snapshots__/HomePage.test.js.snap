// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home-container"
  >
    <h1
      class="Home-title"
    >
      Welcome to Jobly!
    </h1>
    <h6
      class="description text-muted"
    >
      A place you can find and apply to the job of your dreams...
    </h6>
    <div
      class="Login-container"
    >
      <div />
      <div>
        <button
          class="login-signup-btn btn btn-primary"
        >
          Login
        </button>
        <br />
        <button
          class="login-signup-btn btn btn-secondary"
        >
          Register
        </button>
      </div>
      <br />
      <br />
      <form
        class="\\"loginForm\\" \\"form-group\\" Hidden"
      >
        <div
          class="input-group"
        >
          <div
            class="input-group-prepend"
          >
            <span
              class="input-group-text"
            >
              Username
            </span>
          </div>
          <input
            class="form-control"
            name="username"
            value=""
          />
        </div>
        <br />
        <div
          class="input-group"
        >
          <div
            class="input-group-prepend"
          >
            <span
              class="input-group-text"
            >
              Password
            </span>
          </div>
          <input
            class="form-control"
            name="password"
            value=""
          />
        </div>
        <br />
        <button
          class="login-btn btn btn-primary"
          type="submit"
        >
          Submit
        </button>
      </form>
      <form
        class="\\"signUpForm\\" Hidden"
      >
        <div
          class="input-group"
        >
          <div
            class="input-group-prepend"
          >
            <span
              class="input-group-text"
            >
              First and last name
            </span>
          </div>
          <input
            class="form-control"
            name="first_name"
            value=""
          />
          <br />
          <input
            class="form-control"
            name="last_name"
            value=""
          />
          <br />
        </div>
        <br />
        <div
          class="input-group"
        >
          <div
            class="input-group-prepend"
          >
            <span
              class="input-group-text"
            >
              Email
            </span>
          </div>
          <input
            class="form-control"
            name="email"
            value=""
          />
        </div>
        <br />
        <div
          class="input-group"
        >
          <div
            class="input-group-prepend"
          >
            <span
              class="input-group-text"
            >
              Username
            </span>
          </div>
          <input
            class="form-control"
            name="username"
            value=""
          />
        </div>
        <br />
        <div
          class="input-group"
        >
          <div
            class="input-group-prepend"
          >
            <span
              class="input-group-text"
            >
              Password
            </span>
          </div>
          <input
            class="form-control"
            name="password"
            value=""
          />
        </div>
        <button
          class="signup-btn btn btn-secondary"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
